bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/things/VeluxExistingProducts.java

```diff
@@ -123,8 +123,13 @@ public boolean update(ProductBridgeIndex bridgeProductIndex, int productState, i
             return false;
         }
         VeluxProduct thisProduct = this.get(bridgeProductIndex);
-        if (thisProduct.setState(productState) || thisProduct.setCurrentPosition(productPosition)
-                || thisProduct.setTarget(productTarget)) {
+        /*
+         * If the following if statement used the || operator, the compiler would do logical short circuit optimisation
+         * so that if the leftmost function evaluated to true, the further right functions would not get called; so we
+         * use the | operator instead which prevents short circuiting so all three property setters are always called.
+         */
+        if (thisProduct.setState(productState) | thisProduct.setCurrentPosition(productPosition)
+                | thisProduct.setTarget(productTarget)) {
             dirty = true;
```

Comment:
I'd prefer if you avoided all of the boolean operator trickery in general.

```suggestion
        dirty |= thisProduct.setState(productState);
        dirty |= thisProduct.setCurrentPosition(productPosition);
        dirty |= thisProduct.setTarget(productTarget);
        if (dirty) {
```

Reply:
Hmm. The difference between || and | is a well documented (albeit little understood) aspect of Java, so I would not call it 'trickery'. However I will unwrap the single line multi | expression to multi line single |= expressions, if you insist. I am pretty sure the compiler would generate the same underlying executable code anyway.